src/workerd/api/urlpattern-standard.c++

```diff
@@ -0,0 +1,253 @@
+#undef DEFINE_GETTER
+
+jsg::Ref<URLPattern> URLPattern::constructor(jsg::Lock& js,
+    jsg::Optional<kj::OneOf<kj::String, URLPatternInit>> maybeInput,
+    jsg::Optional<kj::OneOf<kj::String, URLPatternOptions>> maybeBase,
+    jsg::Optional<URLPatternOptions> maybeOptions) {
+  ada::url_pattern_input input;
+  std::optional<std::string_view> base{};
+  std::optional<ada::url_pattern_options> options{};
+
+  KJ_IF_SOME(mi, maybeInput) {
+    KJ_SWITCH_ONEOF(mi) {
+      KJ_CASE_ONEOF(str, kj::String) {
+        input = std::string_view(str.begin(), str.size());
+      }
+      KJ_CASE_ONEOF(init, URLPatternInit) {
+        input = init.toAdaType();
+      }
+    }
+  } else {
+    input = ada::url_pattern_init{};
+  }
+
+  KJ_IF_SOME(b, maybeBase) {
+    KJ_SWITCH_ONEOF(b) {
+      KJ_CASE_ONEOF(str, kj::String) {
+        base = std::string_view(str.begin(), str.size());
+      }
+      KJ_CASE_ONEOF(o, URLPatternOptions) {
+        options = o.toAdaType();
+      }
+    }
+  }
+
+  if (!options.has_value()) {
+    KJ_IF_SOME(o, maybeOptions) {
+      options = o.toAdaType();
+    }
+  }
+
+  std::string_view* base_opt = base ? &(*base) : nullptr;
+  auto result =
+      ada::parse_url_pattern<URLPatternRegexEngine>(input, base_opt, options ? &*options : nullptr);
+  if (result.has_value()) {
+    return jsg::alloc<URLPattern>(std::move(*result));
+  }
+
+  JSG_FAIL_REQUIRE(TypeError, "Failed to construct URLPattern");
+}
+
+bool URLPattern::test(jsg::Optional<kj::OneOf<kj::String, URLPatternInit>> maybeInput,
+    jsg::Optional<kj::String> maybeBase) {
+  ada::result<bool> result;
+  std::optional<std::string_view> base{};
+
+  KJ_IF_SOME(b, maybeBase) {
+    base = std::string_view(b.begin(), b.size());
+  }
+
+  std::string_view* base_ptr = base ? &*base : nullptr;
+
+  KJ_IF_SOME(mi, maybeInput) {
+    KJ_SWITCH_ONEOF(mi) {
+      KJ_CASE_ONEOF(str, kj::String) {
+        result = inner.test(std::string_view(str.begin(), str.size()), base_ptr);
+      }
+      KJ_CASE_ONEOF(pi, URLPattern::URLPatternInit) {
+        result = inner.test(pi.toAdaType(), base_ptr);
+      }
+    }
+  } else {
+    result = inner.test(ada::url_pattern_init{}, base_ptr);
+  }
+
+  JSG_REQUIRE(result.has_value(), TypeError, "Failed to test URLPattern");
+
+  return *result;
+}
+
+kj::Maybe<URLPattern::URLPatternResult> URLPattern::exec(jsg::Lock& js,
+    jsg::Optional<kj::OneOf<kj::String, URLPatternInit>> maybeInput,
+    jsg::Optional<kj::String> maybeBase) {
+  ada::result<std::optional<ada::url_pattern_result>> result;
+  std::optional<std::string_view> base_url{};
+  std::string base_url_base;
+
+  KJ_IF_SOME(b, maybeBase) {
+    base_url_base = std::string(b.begin(), b.size());
+    base_url = std::string_view(base_url_base);
+  }
+
+  KJ_IF_SOME(mi, maybeInput) {
+    KJ_SWITCH_ONEOF(mi) {
+      KJ_CASE_ONEOF(str, kj::String) {
+        result = inner.exec(
+            std::string_view(str.begin(), str.size()), base_url ? &(*base_url) : nullptr);
```

Comment:
```suggestion
            std::string_view(str.begin(), str.size()), base_url ? base_url.data() : nullptr);
```
as a general find and replace.
the &* syntax is horrible imho.

Reply:
This parameter has `std::string_view*` type. I don't think this would work because base_url.data() doesn't return std::string_view* nor it can be casted into string_view*. Please correct me if i'm wrong

Reply:
Yeah sorry, didn't notice it's an optional 
Then you could do
'&base_url.value()' 
Still nice imho then the current syntax